workers/event_catcher/event_catcher.rb

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/env ruby
+
+require "manageiq-messaging"
+require "manageiq-password"
+require "pathname"
+
+class EventCatcher
+  def initialize(ems_id, hostname, username, password, port, messaging_host, messaging_port, page_size = 20)
+    @ems_id         = ems_id
+    @hostname       = hostname
+    @username       = username
+    @password       = ManageIQ::Password.try_decrypt(password)
+    @port           = port
+    @messaging_host = messaging_host
+    @messaging_port = messaging_port
+    @page_size      = page_size
+  end
+
+  def run!
+    vim                     = connect
+    event_history_collector = create_event_history_collector(vim, page_size)
+    property_filter         = create_property_filter(vim, event_history_collector)
+
+    wait_for_updates(vim) do |property_change|
+      next unless property_change.name =~ /latestPage.*/
+
+      events = Array(property_change.val).map { |event| parse_event(event) }
+      publish_events(events)
```

Comment:
Would this be better as parse1/publish1/parse2/publish2/... as opposed to parse1/parse2/.../publish1/publish2/... ?

Reply:
I'm hoping to use the "efficient message producer" model where you batch up the set of messages then commit them together: https://github.com/appsignal/rdkafka-ruby#producing-messages

Reply:
Ah ok...didn't realize there was a batch-"put"